test/teyu_init_test.rb

```diff
@@ -0,0 +1,158 @@
+require_relative "test_helper"
+require 'ostruct'
+
+class TeyuInitTest < Test::Unit::TestCase
+  module GenericTeyu
+    def teyu_init(*params)
+      argument = Teyu::Argument.new(params)
+      Teyu::GenericInitializer.new(self, argument).define
+    end
+  end
+
+  data(:module, [GenericTeyu, Teyu], keep: true)
```

Comment:
Now, we test both `PerfectInitializer` and `GenericInitializer` so that we can assure GenericInitializer works fine alone.